### DISCUSSION TITLE
Can I get information about the file at the end of the drive redirection?

### DISCUSSION THREAD

**Discussion by kimcoldwater** [FreeRDP/FreeRDP#8788] [Q&A]
Hello, I'm using freerdp to make rdp functions available on the web.

Creating a file transfer function via drive redirection.

For analysis, we checked the following documents and debugged the sources below: `channels/drive/client/`

`https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/16023025-8a78-492f-8b96-c873b042ac50
`
`http://pub.freerdp.com/api/drive__main_8c.html
`

Please understand even if the question is not accurate due to insufficient analysis.

Q1. When I move files using drive redirection, can I know when the drive redirection ends?
Like `endpaint`.

Q2. If there is an end point, is it possible to take data out of the stream and use it?

I've tried to look at the document and leave a log on all the functions to see how they work, but i still very poor.

I look forward to your advice.

**Comment by akallabeth** [comment]
I don´t exactly know what you´re asking here. Might help if you describe what you actually try to do on a higher level.

**Reply by kimcoldwater** [reply]
When you upload a specific file from the web to a server, the server that is actually running freerdp tries to put the file into the drive-redirected folder.

In addition, when you move a file from a remote session to a redirected folder, you want to obtain information about the file and make it available for Web users to download.

My server connects to the web client - freerdp running server ( java,jni ) - remote session.

To implement this, I'm using drive redirection to check when the file is moved and whether the file's information (such as file name) can be obtained. There are two things that are needed.

If you can get the file data itself within FREERDP, it's even better because you don't need to search for local files.

**Reply by akallabeth** [reply]
ok, so this has more or less nothing to do with drive redirection but only your client.

* you can use a temporary folder locally to have the data stored there. Just use any file notification API available to you, no changes to `FreeRDP` required
* you need to implement your own drive redirection channel to have a virtual filesystem available so you don´t need to store files.

**Reply by kimcoldwater** [reply]
I have implemented this function through your advice.

After implementation, file uploads/downloads found that using SFTP is a way closer to requirements. 

This feature will be used for drive redirection and SFTP will be used for file upload/download. 

Thank you always for your advice.
